### PR TITLE
Reorganize Product/Package to Package/SummaryInformation.

### PR DESCRIPTION
**Pull request by barnson**

### CODE REVIEW

File: src/WixToolset.Core/Compiler_2.cs [hunk 1]
```diff
@@ -315,6 +410,74 @@ private void ParseProductElement(XElement node)
...
+                    msiVersion = 200;
+                    break;
+                case Platform.ARM64:
+                    platform = "Arm64";
+                    msiVersion = 500;
+                    break;
+                default:
+                    throw new ArgumentException("Unknown platform enumeration '{0}' encountered.", this.CurrentPlatform.ToString());
+            }
+
```
Comment (robmen) [inline src/WixToolset.Core/Compiler_2.cs:432]: Unnecessary whitespace?

File: src/WixToolset.Core/Compiler_2.cs [hunk 2]
```diff
@@ -140,6 +180,25 @@ private void ParseProductElement(XElement node)
...
+                {
+                    this.AddProperty(sourceLineNumbers, new Identifier(AccessModifier.Public, "ALLUSERS"), "1", false, false, false, false);
+                }
+
+                this.ValidateAndAddCommonSummaryInformationSymbols(sourceLineNumbers, msiVersion, platform);
+
+                this.Core.AddSymbol(new SummaryInformationSymbol(sourceLineNumbers)
+                {
+                    PropertyId = SummaryInformationType.PackageCode,
+                    Value = "*"
```
Comment (robmen) [inline src/WixToolset.Core/Compiler_2.cs:193]: Should we set this in the backend rather than default to `*`?

File: src/WixToolset.Core/Compiler_2.cs [hunk 3]
```diff
@@ -315,6 +410,74 @@ private void ParseProductElement(XElement node)
             }
         }
 
+        private void GetDefaultPlatformAndInstallerVersion(out string platform, out int msiVersion)
+        {
+            switch (this.CurrentPlatform)
```
Comment (robmen) [inline src/WixToolset.Core/Compiler_2.cs:415]: What if we default `msiVersion = 500;` for everything? It would give a strong message to people that Win7 is min.

**Review by robmen — APPROVED** [review]
Minor things.

File: src/WixToolset.Core/Compiler_2.cs [hunk 4]
```diff
@@ -23,15 +23,24 @@ internal partial class Compiler : ICompiler
...
+            var isPerMachine = true;
+            string installScope = null;
             string upgradeCode = null;
             string manufacturer = null;
             string version = null;
             string symbols = null;
+            var setCodepage = false;
+            var setPackageName = false;
+            var setKeywords = false;
+            var setPackageAuthor = false;
```
Comment (rseanhall) [inline src/WixToolset.Core/Compiler_2.cs:41]: I find `setX` to be ambiguous between "X has been set" and the imperative "[you] set X".

**Review by rseanhall — APPROVED** [review]

File: src/WixToolset.Core/Compiler_2.cs (hunk 3 above)
Comment (barnson) [reply src/WixToolset.Core/Compiler_2.cs:415]: 👍
